Mathlib/Data/Fin/Basic.lean

```diff
@@ -147,6 +147,10 @@ theorem ne_iff_vne (a b : Fin n) : a ≠ b ↔ a.1 ≠ b.1 :=
 theorem mk_eq_mk {a h a' h'} : @mk n a h = @mk n a' h' ↔ a = a' :=
   Fin.ext_iff
 
+lemma val_ne_zero_iff {n : ℕ} {k : Fin (n+1)} :
+  k ≠ 0 ↔ (k : ℕ) ≠ 0 := by
+  rw [←Fin.val_ne_iff, Fin.val_zero]
```

Comment:
This lemma is backwards, `val_ne` refers to the RHS not the LHS.

Can you also add the `val_eq_zero_iff` version, and mark both `norm_cast`?